jerry-core/ecma/operations/ecma-regexp-object.cpp

```diff
@@ -282,8 +319,9 @@ re_match_regexp (re_matcher_ctx_t *re_ctx_p, /**< RegExp matcher context */
           return ecma_make_simple_completion_value (ECMA_SIMPLE_VALUE_FALSE); /* fail */
         }
 
-        ecma_char_t ch1 = (ecma_char_t) re_get_value (&bc_p);
-        ecma_char_t ch2 = lit_utf8_iterator_read_next (&iter);
+        bool is_ignorecase = re_ctx_p->flags & RE_FLAG_IGNORE_CASE;
+        ecma_char_t ch1 = re_canonicalize ((ecma_char_t) re_get_value (&bc_p), is_ignorecase);
+        ecma_char_t ch2 = re_canonicalize (lit_utf8_iterator_read_next (&iter), is_ignorecase);
```

Comment:
Should affect the compile, not just the match. The compiled regex should only have uppercase characters.

Casesless backreferences are supported correctly?